personal_graph/natural.py

```diff
@@ -16,19 +16,30 @@
 
 load_dotenv()
 
-if os.getenv("OPEN_API_KEY"):
-    client = instructor.patch(OpenAI(api_key=os.getenv("OPEN_API_KEY")))
+if os.getenv("LITE_LLM_BASE_URL"):
+    openai_client = openai.OpenAI(
+        api_key="",
+        base_url=os.getenv("LITE_LLM_BASE_URL"),
+        default_headers={"Authorization": f"Bearer {os.getenv('LITE_LLM_TOKEN')}"},
+    )
+    client = instructor.from_openai(openai_client)
 else:
-    client = None
+    client = None  # type: ignore
 
 db_url = os.getenv("LIBSQL_URL")
 auth_token = os.getenv("LIBSQL_AUTH_TOKEN")
 
 
 def generate_graph(query: str) -> KnowledgeGraph:
-    client = instructor.patch(OpenAI(api_key=os.getenv("OPEN_API_KEY")))
+    client = instructor.from_openai(
```

Comment:
you should refactor this part out, currently it leads to unnecessary code duplication

Reply:
Done